toolkit/utils/ogmios-client/src/jsonrpsee.rs

```diff
@@ -1,36 +1,72 @@
 //! OgmiosClient implementation with jsonrpsee.
 //! Major drawback is that it swallows the error response from the server in case of 400 Bad Request.
 
+use std::io::Write;
+
 use crate::{OgmiosClient, OgmiosClientError, OgmiosParams};
-use jsonrpsee::core::params::ArrayParams;
 use jsonrpsee::{
-	core::{client::ClientT, params::ObjectParams},
+	core::{client::ClientT, traits::ToRpcParams, ClientError},
 	http_client::HttpClient,
 };
 use serde::de::DeserializeOwned;
 
+const CONTRACTLOG_FILE: &str = "contractlog.json";
+
+macro_rules! contractlog {
+	($($arg:tt)*) => {
+		add_to_contractlog(&format!("{}\n", format!($($arg)*)))
```

Comment:
I expected to see `log::info!` and the user - smart contract commands, maybe pc-cli wizards setting logger to file.

Reply:
Also, `contractlog.json` name was leaking technical detail - Contract monad from CTL. We can do something less traumatic.

Reply:
Yeah, good idea. `env_logger` doesn't support that, but `log4rs` turns out to be very nice to use.